coremark/01-core_portme.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/posix/core_portme.mak b/posix/core_portme.mak
+index 2852069..3eaaabe 100755
+--- a/posix/core_portme.mak
++++ b/posix/core_portme.mak
+@@ -24,8 +24,12 @@ OUTFLAG= -o
+ CC?= cc
+ # Flag: CFLAGS
+ #	Use this flag to define compiler options. Note, you can add compiler options from the command line using XCFLAGS="other flags"
+-PORT_CFLAGS = -O2
+-FLAGS_STR = "$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END)"
++PORT_CFLAGS =
++FLAGS_STR := $(filter-out -I%,$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END))
++FLAGS_STR := $(filter-out --sysroot=%,$(FLAGS_STR))
```

Comment:
uh, removing the `--sysroot` makes the port build against `libphoenix` embedded in toolchain, not in project - so IMHO not intended? Also `-B` is needed for selection correct `crt0.o` (project against toolchain) if I remember correctly

Reply:
This is only for the `FLAGS_STR` define (which prints out in coremark)
```
CoreMark Size    : 666
Total ticks      : 21151
Total time (secs): 21.151000
Iterations/Sec   : 520.069973
Iterations       : 11000
Compiler version : GCC9.5.0
Compiler flags   : -fomit-frame-pointer -mcmodel=medany --sysroot=/home/lukasz/master-phoenix/phoenix-rtos-project/_build/riscv64-generic-qemu/sysroot/ -B/home/lukasz/master-phoenix/phoenix-rtos-project/_build/riscv64-generic-qemu/sysroot/lib/ -iprefix /home/lukasz/master-phoenix/phoenix-rtos-project/_build/riscv64-generic-qemu/sysroot/ -fmacro-prefix-map=/home/lukasz/master-phoenix/= -ffunction-sections -fdata-sections -DUSE_PTHREAD -DMULTITHREAD=1 -L/home/lukasz/master-phoenix/phoenix-rtos-project/_build/riscv64-generic-qemu/lib/ -Wl,--gc-sections
```
I removed the flags from the string so that we don't get wall of flags not affecting performance of the benchmark.
The application if built with all the flags defined.

Reply:
see https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/makes/setup-sysroot.mk#L20 for detailed breakdown of how sysroot options are used.

Reply:
Hmm I don't think we understand each other :) `FLAGS_STR` is only provided to preprocessor define (https://github.com/eembc/coremark/blob/d5fad6bd094899101a4e5fd53af7298160ced6ab/posix/core_portme.mak#L29). Here's what actual build uses: https://github.com/eembc/coremark/blob/d5fad6bd094899101a4e5fd53af7298160ced6ab/Makefile#L87

Only I've now realized that `XLFLAGS` are not passed correctly, will fix that.

Reply:
sorry, when I was posting my reply, I haven't yet see your reply (it was just an update over original comment).

I was mislead by the original comment above (that it's being used for compilation) and the patch doesn't have the variable usage. Maybe small comment regarding why you're removing it / it's safe to do so would prevent any confusion in the future ;)